wordpress-mobile/WordPress-iOS
# Reading Preferences: Apply styling to toolbar and navigation bar

Part of #22925 
Depends on #22929

This PR applies the reading preferences style to the Reader detail toolbar and the navigation bar. Some previews:

| ![Simulator Screenshot - iPhone 15 - 2024-04-02 at 20 28 28](https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/47a2f13f-dbda-4a7d-87b1-ba263538bd91) | ![Simulator Screenshot - iPhone 15 - 2024-04-02 at 20 28 14](https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/f5ad9527-49a0-4cb5-a1af-40a0ed4194bc)
-|-

Apart from styling, some there are some additional updates:

- `useCompatibilityMode` is assumed `false` when the feature flag is active. It seems that the purpose of this compatibility mode is for cases where we were still displaying screens with a blue navigation bar color, which was a legacy behavior. Based on the documentation, it seems like we should be able to deprecate this property.
- The `ReaderDetailFeaturedImageView` directly controls the Reader detail's navigation bar style. Particularly, when a featured image exists, we start off with a white tint color for the navigation bar, and as the user scrolls past the featured image, we transition the color to the primary/text color. This behavior only applies when the app is in light scheme.
  - With the customization coming into play, we now need to detect whether the current color theme has a light-colored background.

## To test

- Launch the Jetpack app.
- Ensure that the `Reader Customization` feature flag is turned on. 
- Navigate to the Reader tab, and open any post.
- Update the reading preferences.
- 🔎 Verify that the navigation bar and the toolbar are updated according to the selected preferences.
- Navigate to Reader Detail from other entry points:
  - Open a mention notification.
  - Tap a deeplink to a Reader Post.
- 🔎 Verify that the navigation bar and the toolbar are displayed according to the selected preferences.

Additional: also test while changing the app appearance, for example:
- Sepia with dark mode
- OLED with light mode
- ...etc.

## Regression Notes
1. Potential unintended areas of impact

    - The `useCompatibilityMode` is assumed `false` when the feature flag is on. This means that entry points to the Reader Detail, other than the Reader stream, will have the style applied to the entire screen. 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

3. What automated tests I added (or what prevented me from doing so)
N/A.

PR submission checklist:


Testing checklist:
